packages/techdocs-common/src/stages/publish/awsS3.test.ts

```diff
@@ -46,7 +45,7 @@ const getEntityRootDir = (entity: Entity) => {
     kind,
     metadata: { namespace, name },
   } = entity;
-  const entityRootDir = path.join(namespace as string, kind, name);
+  const entityRootDir = `${namespace}/${kind}/${name}`;
```

Comment:
just noting that it's not particularly safe to assume that these three pieces only contain filesystem safe characters :)

Reply:
Curious what are the allowed characters for namespace, kind and name? TechDocs stores files in cloud storage and local, creating a nested directory like -> `namespace/kind/name/index.html`

Reply:
It's actually [configurable](https://github.com/backstage/backstage/blob/master/plugins/catalog-backend/src/service/CatalogBuilder.ts#L170).

But yeah we can just hope that nobody ever starts allowing slashes and colons and nulls and stuff inside their fields, because that is bound to break stuff all over the place.

Reply:
Since you're talking about paths, do we fix the 
```ts
const relativeFilePath = filePath.replace(`${directory}/`, '');
``` 
line (present in the s3 and gcs publishers) I mentioned in the PR description above, which I think is a problem for Windows users.

Reply:
Hard to say without a windows machine in front of me, but it should probably be `path.relative(directory, filePath)` and handling leading slashes or dots accordingly.

Reply:
As I think this problem is out of scope, I will create an issue to report the problem after re-testing on Windows.